e2etest/bgptests/unnumber.go

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package bgptests
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	frrk8sv1beta1 "github.com/metallb/frr-k8s/api/v1beta1"
+
+	"go.universe.tf/e2etest/pkg/executor"
+	"go.universe.tf/e2etest/pkg/frr"
+	frrconfig "go.universe.tf/e2etest/pkg/frr/config"
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+	"go.universe.tf/e2etest/pkg/k8s"
+	"go.universe.tf/e2etest/pkg/k8sclient"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var _ = ginkgo.FDescribe("FRRK8S-MODE DUALSTACK - Unnumbered BGP", func() {
+	defer ginkgo.GinkgoRecover()
+	var (
+		testNamespace string
+		peers         []*frrcontainer.FRR
+	)
+
+	ginkgo.BeforeEach(func() {
+		c := frrcontainer.Config{
```

Comment:
the goal is to have this as part of the infra setup to make sure everything works with unnumbered?

Reply:
My preference would be to create the new container the unnumber peering as part of the test, to avoid having a container running for the full rutime of the tests. Do you see a reason to move it to the infra setup?

Reply:
> Do you see a reason to move it to the infra setup?

to run all the existing tests against the "unnumbered" scenario (unless we decide we're good with having only a single test for unnumbered)

Reply:
I see your point, but "all existing" means double the time and mabye adding complexity in the setupinfra. But no strong opinion, we can come back to that  on a later step.

Reply:
> means double the time

not sure about that, it would just be another container as part of the tests

> mabye adding complexity

probably, this pr is a good place to understand if it's reasonable or not :)